### PR TITLE
Filter out enabled repos

### PR DESCRIPTION
**Pull request by earnubs**
Enabled repos should be dimmed, inputs checked and disabled.

### CODE REVIEW

File: src/common/components/select-repository-list/index.js [hunk 1]
```diff
@@ -86,20 +87,46 @@ class SelectRepositoryList extends Component {
...
     const { selectedRepos } = this.props.selectRepositoriesForm;
-    const { repos } = this.props.repositories;
+    const { repos, success } = this.props.repositories;
     const pageLinks = this.renderPageLinks.call(this);
 
+    this.filterEnabledRepos(repos);
+    let renderedRepos = null;
+
+    if (success) {
+      renderedRepos = this.props.repositories.repos.map(this.renderRepository.bind(this));
```
Comment (earnubs) [inline src/common/components/select-repository-list/index.js]: not necessary for this PR, but seemed like an OK improvement to make while here

File: src/common/components/select-repository-row/index.js [hunk 2]
```diff
@@ -8,20 +9,29 @@ class SelectRepositoryRow extends Component {
...
             type="checkbox"
             onChange={ this.onChange.bind(this) }
-            checked={ checked }
+            checked={ checked || isEnabled }
+            disabled={ isEnabled }
           />
         }
         <div>
-          {repository.fullName}
+          <label htmlFor={ repository.fullName }>{repository.fullName}</label>
```
Comment (earnubs) [inline src/common/components/select-repository-row/index.js]: fixes #234

File: src/common/components/select-repository-row/index.js [hunk 3]
```diff
@@ -37,12 +47,18 @@ class SelectRepositoryRow extends Component {
   }
 }
 
+SelectRepositoryRow.defaultProps = {
+  checked: false,
```
Comment (earnubs) [inline src/common/components/select-repository-row/index.js]: if the props on a form switch from undef to bool, react warns about controlled vrs uncontrolled form elements.

**Comment by earnubs** [comment]
[[add new repo] filter out repositories that are already enabled](https://trello.com/c/jdl05qOn/174-add-new-repo-filter-out-repositories-that-are-already-enabled)

File: src/common/components/select-repository-list/index.js [hunk 4]
```diff
@@ -86,20 +87,46 @@ class SelectRepositoryList extends Component {
     this.props.dispatch(fetchUserRepositories(pageNumber));
   }
 
+  filterEnabledRepos(repositories) {
+    const { success, snaps } = this.props.snaps;
+
+    if (success && snaps.length) {
+      for (let i = repositories.length; i--;) {
+        for (let j = snaps.length; j--;) {
+          let enabledRepo = snaps[j].git_repository_url;
```
Comment (cjwatson) [inline src/common/components/select-repository-list/index.js]: `const`, I think.

Comment (earnubs) [inline src/common/components/select-repository-list/index.js]: ta

**Pull request merged**